juju/juju
# audit-log: Add auditconfigupdater to machine manifolds

## Description of change

The auditconfigupdater was added to the 2.3 branch but wasn't running in the 2.4 branch because the API server has been moved into the dependency engine. This adds a manifold for the auditconfigupdater, adds the worker to the machine agent manifolds, and updates the API server worker and manifold to receive the GetAuditConfig func as an input.

## QA steps

Bootstrap a new controller with auditing enabled. Check that messages are written to the audit log. Turn off auditing `juju controller-config auditing-enabled=false`. See that the update is mentioned in the debug log, and that messages for new commands aren't written to the audit log. Re-enable auditing and see that new commands are added again. Check that upgrading juju works, and that changing the controller config can be done afterwards.
